addons/mail/models/mail_thread.py

```diff
@@ -694,6 +694,10 @@ def _notify_get_action_link(self, link_type, **kwargs):
             params['token'] = token
 
         link = '%s?%s' % (base_link, url_encode(params))
+
+        if hasattr(self, 'get_base_url'):
```

Comment:
Not sure which syntax is best but now that I think about it we sometimes also write `if 'get_base_url' in self:`

Reply:
I will keep it that way as this whole file mainly use `hasattr`

Reply:
Plus `'method' in self` seems to not be working with methods but only with fields/properties

Reply:
Well spotted, indeed `__contains__` in `Model` looks into `_fields` only.